nix/output.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, name ? "default-output", ... }:
+with lib;
+{
+  options = {
+    name = mkOption {
+      default = "${name}";
+      type = types.str;
+      description = "Name of the output.";
+    };
+    func = mkOption {
+      default = null;
+      type = types.nullOr (types.str);
+      #type = types.nullOr (types.either types.str types.path);
+      description = ''
+        Text of a shell script which will produce a JSON value.
```

Comment:
I wonder if we should limit this to shell. I could imagine with a high-level programming languages
it would be easier to generate json in a safe way. Also one could leverage libraries.